src/services/aggregatedLoggingService.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+angular.module("openshiftCommonServices")
+  .factory("AggregatedLoggingService", function($q, $cacheFactory, Logger, $interval, APIService, DataService) {
+
+    var currentUser = null;
+    var cachedUserPermissions = $cacheFactory('operationsUsersCache', {
+          number: 10
+        });
+
+    inFlightPermissionsRequests = {};
+
+    // Create SelfSubjectAccessReview request againt authorization API to check whether
+    // current user can view pods/log from 'openshift-logging' project.
+    // Users who can view such logs are 'operations' users
+    var fetchUserPermisions = function(userName) {
+      var deferred = $q.defer();
```

Comment:
You shouldn't need to create a new deferred. Just return the promise from `DataService.create()` or return `$q.when(cachedValue)` when returning a cached value.